tests/nvmeof/workflows/initiator.py

```diff
@@ -14,6 +14,32 @@ def __init__(self, node, gateway):
         self.gateway = gateway
         self.discovery_port = 8009
 
+    def fetch_nvme_devices_for_gw(self, gateway):
+        """Fetch nsids for all devices optimised for the given gateway
+
+        Returns:
+            nsids for which the given gateway is optimised.
+            empty list if the gateway is inaccessible for all devices
+        """
+        out, _ = self.node.exec_command(
+            cmd="nvme list --output-format json", sudo=True
+        )
+        nvme_devices = json.loads(out).get("Devices", [])
+        gw_ip = gateway.node.ip_address
+        nsids = []
+        for nvme_device in nvme_devices:
+            nsid = nvme_device.get("NameSpace")
+            device = nvme_device.get("DevicePath")
+            out, _ = self.node.exec_command(
+                cmd=f"nvme list-subsys {device} --output-format json", sudo=True
+            )
+            subsystems = json.loads(out)[0].get("Subsystems")
+            for subsys in subsystems:
+                for path in subsys.get("Paths"):
+                    if gw_ip in path.get("Address") and path.get("State") == "live" and path.get("ANAState") == "optimized":
```

Comment:
create separate method to list out optimized and inaccessible paths for a device.
`{"device_wwn": {"optimized": []}, {"inaccessible": []}`. use it for validation of paths

Reply:
will do this change as well.

Reply:
Done.